### ISSUE TITLE
findDatesUTC not working

### ISSUE THREAD

**Issue by FernetB** [sentinel-hub/sentinelhub-js#148]
Hi, me again! xD

I keep getting this error when i tried to use the `findDatesUTC` method

I don't know what is the Dimension field that is missing I didn't saw it in the docs.

![errorsentinel](https://user-images.githubusercontent.com/47339424/101662588-74f13600-3a28-11eb-80d6-646024874dc8.jpg)

**Comment by sinergise-anze** [comment]
Hi,

Sorry for a bit longer explanation, but maybe it helps someone else in the future too. :slightly_smiling_face: 

**TL;DR**: don't use `WmsLayer`, use an appropriate layer like `S2L2ALayer`. To make it, use this code:
```
  // this creates appropriate layer object with full capabilities it can use:
  const layer = await LayersFactory.makeLayer('https://services.sentinel-hub.com/ogc/wms/<your-instance-id>', '<layer-id>');
```

You will also need to [set authentication token](https://github.com/sentinel-hub/sentinelhub-js#authentication-for-processing-api) first.

**Longer explanation:**

`WmsLayer` is exactly what is in its name - an OGC WMS capable layer. You can turn it towards any WMS service and it should work with it. However, WMS standard does not provide a way to search for the available capture times for a selected area. There is `Dimension` field in `GetCapabilities` response, but it returns *all* the matching dates (/times) for any area, which is not very useful in case of Sentinel Hub - the data is available for every recent enough time interval, just not necessarily for the selected area of interest.

For this reason Sentinel Hub uses other services to get the data (like [Catalog API](https://docs.sentinel-hub.com/api/latest/api/catalog/). If you would like to use these services however, you need to use dataset-specific layers - `WmsLayer` was not built with them in mind.

You can create these layers manually (`layer = new S2L2ALayer(...)` - see [Storybook examples](https://github.com/sentinel-hub/sentinelhub-js/tree/master/stories)). 

Alternatively, if you still want to lean on the layers configured in Sentinel Hub Dashboard, there is a helper method available which makes creating such layers easier:
```
  const layer = await LayersFactory.makeLayer('https://services.sentinel-hub.com/ogc/wms/<your-instance-id>', '<layer-id>');
```

You can use such layers in the same way as `WmsLayer`, except that `findDates` (and similar) methods will now know that they should use appropriate APIs.

Note that you will almost certainly need to [set authentication token](https://github.com/sentinel-hub/sentinelhub-js#authentication-for-processing-api) first, otherwise you probably will not have permissions to get the information about these layers (or to perform Catalog API queries).

**Comment by FernetB** [comment]
Really big thank you!

The problem was solve with this.

Love u for take your time with this full detailed response.

Cheers and Merry Christmas!

**Comment by sinergise-anze** [comment]
No problem, happy to help.

Merry Christmas to you too!
